src/grid.js

```diff
@@ -0,0 +1,145 @@
+var Grid = function(content) {
+  this.cells = {};
+  this.rows = {};
```

Comment:
I think the grid is doing too much, solving is not the responsibilty of the grid it should be the responsibility of another module, maybe `Game` or `Solver` or something